mds/src/karatsuba_convolution.rs

```diff
@@ -0,0 +1,346 @@
+//! Calculate the convolution of two vectors using a Karatsuba-style
+//! decomposition and the CRT.
+//!
+//! This is not a new idea, but we did have the pleasure of
+//! reinventing it independently. Some references:
+//! - https://cr.yp.to/lineartime/multapps-20080515.pdf
+//! - https://2π.com/23/convolution/
+//!
+//! Given a vector v \in F^N, let v(x) \in F[X] denote the polynomial
+//! v_0 + v_1 x + ... + v_{N - 1} x^{N - 1}.  Then w is equal to the
+//! convolution v * u if and only if w(x) = v(x)u(x) mod x^N - 1.
+//! Additionally, define the "signed convolution" by w(x) = v(x)u(x)
```

Comment:
Turns out this has a name! It's a _negacyclic convolution_.

Reply:
You can actually do it with an FFT by multiplying the FFT inputs/outputs by particular factors, but I'd have to re-derive it to give an exact method. (This came up when I was doing Poseidon work like 2 years ago.)